lib/cacheWrapper.js

```diff
@@ -1,76 +1,116 @@
 'use strict';
 
 var cacheWrapper = module.exports = {};
-
-var _ = require( 'lodash' );
-var Catbox = require( 'catbox' );
-var catboxMemory = require( 'catbox-memory' );
-var catboxRedis = require( 'catbox-redis' );
-var Q = require( 'q' );
-
+var _ = require('lodash');
+var Catbox = require('catbox');
+var catboxRedis = require('catbox-redis');
+var redisWrapper = require('./redisWrapper');
+var Q = require('q');
 var cacheClient;
 var startingClient = false;
 var queryQueue = [];
 var policies = [];
+var partition = '';
+
+cacheWrapper.initialise = function(serverConfig, cachePolicies) {
+  // no config, no cache...
+  if (_.isEmpty(serverConfig) || _.isEmpty(serverConfig.partition)) {
+    return Q.reject('server config missing partition');
+  }
+  if (_.isEmpty(cachePolicies)) {
+    return Q.reject('cachePolicies missing options(segment/expiresIn)');
+  }
+  partition = serverConfig.partition;
 
-cacheWrapper.initialise = function( serverConfig, cachePolicies ) {
   var deferred = Q.defer();
-  // if serverConfig is empty use an in-memory version of the cache
-  var catboxType = _.isEmpty( serverConfig ) ? catboxMemory : catboxRedis;
-  cacheClient = new Catbox.Client( catboxType, serverConfig );
 
+  cacheClient = new Catbox.Client(catboxRedis, serverConfig);
   // Build all of the cache policies into the client
-  _.each( cachePolicies, function( cachePolicy ) {
-    policies[cachePolicy.segment] = new Catbox.Policy( { expiresIn: cachePolicy.expiresIn }, cacheClient, cachePolicy.segment );
-  } );
+  _.each(cachePolicies, function(cachePolicy) {
+    if (!_.isUndefined(cachePolicy.segment) && !_.isUndefined(cachePolicy.expiresIn)) {
+      policies[cachePolicy.segment] = new Catbox.Policy(
+        {
+          expiresIn: cachePolicy.expiresIn
+        },
+        cacheClient, cachePolicy.segment
+      );
+    }
+  });
 
   // fire up the client
-  cacheClient.start( function() {
-    deferred.resolve();
-  } );
+  cacheClient.start(function() {
+    redisWrapper.initialise(serverConfig)
+    .then(function() {
+      deferred.resolve();
+    })
+    .fail(function() {
+      deferred.resolve('redis client was not initialised');
```

Comment:
we will want the systems requiring cache to still continue working without the cache if there are issues with the cache that is why I have resolved it will add a comment